lib/datasource/docker/index.ts

```diff
@@ -56,7 +59,10 @@ async function getAuthHeaders(
       headers?: Record<string, string>;
     } = hostRules.find({ hostType: 'docker', url: apiCheckUrl });
     opts.json = true;
-    if (opts.username && opts.password) {
+    if (ecrRegex.test(registry)) {
+      const auth = await getECRAuthToken(registry.match(ecrRegex)[1], opts);
+      opts.headers = { authorization: `Basic ${auth}` };
+    } else if (opts.username && opts.password) {
```

Comment:
Is there any circumstance where our existing username/password implementation *would* work for ECR? I want to make sure we're not doing anything backwards-incompatible because this is now inside an `else if`

Reply:
Not to my understanding. See https://docs.aws.amazon.com/AmazonECR/latest/APIReference/API_GetAuthorizationToken.html